packages/rest-generator/src/parser/refs.ts

```diff
@@ -0,0 +1,36 @@
+import { last, pascalCase } from '@sap-cloud-sdk/util';
+import { OpenAPIV3 } from 'openapi-types';
+import { $Refs } from '@apidevtools/swagger-parser';
+
+/**
+ * Typeguard to check wheter an object is of type `OpenAPIV3.ReferenceObject`.
+ * @param obj Object to check.
+ * @returns True if the object is a reference object, false otherwise.
+ */
+export function isReferenceObject(obj: any): obj is OpenAPIV3.ReferenceObject {
+  return !!obj?.$ref;
+}
+
+/**
+ * Parse the type name of a reference object.
+ * @param obj Reference object to get the type name from.
+ * @returns Parsed type name.
+ */
+export function parseTypeName(obj: OpenAPIV3.ReferenceObject): string {
+  // TODO: How do we know that this is correct?
+  return pascalCase(last(obj.$ref.split('/'))!);
+}
+
+/**
+ * Check whether the given object is a reference object and resolve if necessary.
+ * This operates only on the current level.
+ * @param obj Object to resolve if necessary.
+ * @param refs References to resolve by.
+ * @returns A resolved object.
+ */
+export function resolveObject<T>(
+  obj: T | OpenAPIV3.ReferenceObject,
+  refs: $Refs
+): T {
+  return isReferenceObject(obj) ? refs.get(obj.$ref) : obj;
```

Comment:
does this `get` also handle layered resolution so if the ref consists of other refs consistings of refs... until you reach the actual definition. Also cyclic references could be an issue there.

Reply:
It does not handle layered resolution (so also no cyclic refs). Maybe I should improve the docs a bit here.

Reply:
This test implicitly proves this: https://github.com/SAP/cloud-sdk-js/blob/ec895b4c3387309fb22e7919cdb3e4c3306225ba/packages/rest-generator/src/parser/operation.spec.ts#L89

The docs of this function aren't very helpful with this: https://apitools.dev/swagger-parser/docs/refs.html#getref-options

Reply:
Improved docs.